File: ios/entra_external_id/Sources/entra_external_id/EntraExternalIdPlugin.swift
import Flutter

public class EntraExternalIdPlugin: NSObject, FlutterPlugin, NativeAuthHostApi {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = EntraExternalIdPlugin()
    NativeAuthHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance)
  }

  func getNativeSdkStatus() throws -> NativeSdkStatusMessage {
    NativeSdkStatusMessage(platform: .ios, linked: false, sdkVersion: nil)
  }
}

File: android/src/main/kotlin/io/github/wreos/entra_external_id/EntraExternalIdPlugin.kt
package io.github.wreos.entra_external_id

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Pigeon host implementation for the Entra External ID plugin. */
class EntraExternalIdPlugin : FlutterPlugin, NativeAuthHostApi {
    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        NativeAuthHostApi.setUp(flutterPluginBinding.binaryMessenger, this)
    }

    override fun getNativeSdkStatus() =
        NativeSdkStatusMessage(
            platform = NativePlatformMessage.ANDROID,
            linked = false,
            sdkVersion = null,
        )

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        NativeAuthHostApi.setUp(binding.binaryMessenger, null)
    }
}
